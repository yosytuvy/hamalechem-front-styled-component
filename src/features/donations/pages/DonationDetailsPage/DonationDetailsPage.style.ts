import styled from "styled-components";

export const DonationCard = styled.div`
    margin-top: 5px;
    width: 100%;
    display: flex;
    flex-direction: row-reverse;
    flex-grow: 1;
    background: #656318;
    box-shadow: 0 3rem 6rem rgba(0, 0, 0, 0.3);
`;

export const DonationImg = styled.img`
    width: 60%;
`;

export const DonationContent = styled.div`
    color: #ddd;
    padding: 2rem;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    width: 100%;
`;

export const DonationTitle = styled.h1`
    margin-top: 1rem;
    font-size: 3rem;
    font-weight: 300;
`;

export const DonationDesc = styled.h2`
    font-size: 2rem;
    color: burlywood;
    font-weight: normal;
`;

export const DonationPrice = styled.p`
    font-size: 5rem;
    font-weight: 300;
    margin: 0;
`;


