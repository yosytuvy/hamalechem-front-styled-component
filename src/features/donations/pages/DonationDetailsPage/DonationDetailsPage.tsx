import { DonationCard, DonationContent } from "./DonationDetailsPage.style";
import CardImage from "../../components/CardImage";
import CardHeader from "../../components/CardHeader";
import CardDetails from "../../components/CardDetails";

const fakeData = {
    category: "clothing",
    location: "haifa",
    publisher: "someone",
    details: "Gently used blankets in good condition",
    Flexible: "yes",
};

export const Donation = () => {
    return (
        <DonationCard>
            <CardImage
                imageAlt="cotton blankets"
                imageSrc="/Authenticity50_Cotton_Blankets.webp"
            />
            <DonationContent>
                <CardHeader title="Blankets" />
                <CardDetails content={fakeData} />
            </DonationContent>
        </DonationCard>
    );
};
