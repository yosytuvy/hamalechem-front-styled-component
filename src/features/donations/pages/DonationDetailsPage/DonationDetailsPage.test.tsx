import { render, screen } from '@testing-library/react';
import { Donation } from './DonationDetailsPage';

describe('Donation', () => {

  test('renders CardImage with correct props', () => {
    render(<Donation />);
    const image = screen.getByAltText('cotton blankets'); 
    expect(image).toHaveAttribute('src', '/Authenticity50_Cotton_Blankets.webp');
  });

  test('renders CardHeader with correct title', () => {
    render(<Donation />);
    const title = screen.getByRole('heading', { name: 'Blankets' });
    expect(title).toBeInTheDocument();
  });

  test('renders CardDetails with correct content', () => {
    render(<Donation />);
    expect(screen.getByText(/category/i)).toBeInTheDocument();
    expect(screen.getByText(/clothing/i)).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Donation />);
    expect(asFragment()).toMatchSnapshot();
  });

});