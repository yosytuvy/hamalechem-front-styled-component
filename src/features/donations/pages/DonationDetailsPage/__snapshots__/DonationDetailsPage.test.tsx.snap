// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Donation > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXGCL jvhddM"
  >
    <img
      alt="cotton blankets"
      class="sc-gswONk hIsuie"
      src="/Authenticity50_Cotton_Blankets.webp"
    />
    <div
      class="sc-dkrGBB hzkIPU"
    >
      <h1
        class="sc-hLBatu cDmtPE"
      >
        Blankets
      </h1>
      <div>
        <h2
          class="sc-eDvShL jsfACy"
        >
          <b>
            category : 
          </b>
          clothing
        </h2>
        <h2
          class="sc-eDvShL jsfACy"
        >
          <b>
            location : 
          </b>
          haifa
        </h2>
        <h2
          class="sc-eDvShL jsfACy"
        >
          <b>
            publisher : 
          </b>
          someone
        </h2>
        <h2
          class="sc-eDvShL jsfACy"
        >
          <b>
            details : 
          </b>
          Gently used blankets in good condition
        </h2>
        <h2
          class="sc-eDvShL jsfACy"
        >
          <b>
            Flexible : 
          </b>
          yes
        </h2>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
